### ISSUE TITLE
Not able to sync with custom IMAP server

### ISSUE THREAD

**Issue by Nelyah** [Foundry376/Mailspring#367]
I am unable to sync with the custom IMAP server from my university. 
I get an error message stating:
```
Encountrered an error while syncing *****@******
Try Again
```

I have tried to remove the account, then add it again with no avail. 
I get this issue on my two Linux Manjaro PC. 
The sync with my gmail account works well. 

Is this a bug? 
I pasted below some output of the log file in `.config/Mailspring/mailsync*.log`

```
24921 [2017-11-27 11:08:58.484] [main] [info] ------------- Starting Sync (*************@********) ---------------
24921 [2017-11-27 11:08:58.487] [background] [info] Marking all folders as `busy`
24921 [2017-11-27 11:08:58.487] [metadata] [info] Metadata delta stream starting...
24921 [2017-11-27 11:08:58.489] [background] [info] Syncing folder list...
24921 [2017-11-27 11:09:00.992] [background] [info] Syncing folder list...
24921 [2017-11-27 11:09:01.136] [background] [critical] 
 ***
 *** A string exception occurred during program execution: 
 *** blow up the world
 ***

24921 [2017-11-27 11:09:01.157] [background] [critical]  *** Stack trace (line numbers are approximate):
 *** (unknown)  clone
 ***
```

##### Are there any related issues?
Seems to be the same as:
error while syncing - #357 

##### What operating system are you using?
Linux, up to date Manjaro. 

##### What version of Mailspring are you using?
Version 1.0.8-5a837d0e

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No, I have none.

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
No, this is an issue with a custom server.

##### Is the issue reproducible with a particular attachment, message, signature, etc?
No, I do not use any signature. All I try to do is to sync my account.

--

##### Does this bug exist in another mail client or tool you use?
No. I am using Thunderbird just fine, and everything is working without issue. The issue is present on two different PC (both running Linux Manjaro).

**Comment by bengotow** [comment]
Hey! Thanks for reporting this - this happens when Mailspring detects that the IMAP `uidvalidity` value has changed on your server. It's a rare but valid scenario and we need to handle it (rather than just killing the sync worker), but it's odd that you'd see it right when starting sync. Was Mailspring working OK for a while before this happened?

**Comment by Nelyah** [comment]
Hi, 
Yes, it did work for a while before, and I was able to send mails. Since this issue arose however, I have not been able to send any mails. 
It is able -sometimes- to download a few mails before encountering that error. It has for example been able to detect that I created a few upstream folders (although it did not sync the mails in it).

**Comment by Nelyah** [comment]
Hey! Just dropping an update:
I updated today to the new version `1.0.9-c5e85dc4` and it actually fixed my issue! Was it intended? 

Anyway, thanks for the update, I'll be happily sending mails all over again.

**Comment by cogeorg** [comment]
Hi, I still have this issue when trying to use mailspring with protonmail. See log below. Any help greatly appreciated.

1 STARTTLS
1 OK Begin TLS negotiation now
�뵓�2 LOGIN "XXXX@protonmail.com" *********
* CAPABILITY IMAP4rev1 AUTH=LOGIN AUTH=PLAIN ID XSTOP APPENDLIMIT
2 OK [CAPABILITY IMAP4rev1 IDLE MOVE SPECIAL-USE ID XSTOP QUOTA APPENDLIMIT=26214400] LOGIN completed
3 LIST "" ""
* LIST (\Noselect) "/" "/"
3 OK LIST completed
4 ID NIL
* ID (support-url https://protonmail.com/support name ProtonMail vendor "Proton Technologies AG")
4 OK ID completed
ID (support-url https://protonmail.com/support name ProtonMail vendor "Proton Technologies AG")
5 LIST "" "*"

**Comment by stellarsquall** [comment]
Same issue attempting to use ProtonMail Bridge. Adding as a custom IMAP server it appears the authentication is at least working correctly when the bridge is running, but I'm unable to add the account.

`----------IMAP----------
* OK [CAPABILITY IMAP4rev1 STARTTLS AUTH=PLAIN AUTH=LOGIN ID APPENDLIMIT] IMAP4rev1 Service Ready
1 STARTTLS
1 OK Begin TLS negotiation now
2 LOGIN "photosphere@protonmail.ch" *********
* CAPABILITY IMAP4rev1 AUTH=PLAIN AUTH=LOGIN ID XSTOP APPENDLIMIT
2 OK [CAPABILITY IMAP4rev1 IDLE MOVE SPECIAL-USE ID XSTOP QUOTA APPENDLIMIT=26214400] LOGIN completed
3 LIST "" ""
* LIST (\Noselect) "/" "/"
3 OK LIST completed
4 ID NIL
* ID (support-url https://protonmail.com/support name ProtonMail vendor "Proton Technologies AG")
4 OK ID completed
ID (support-url https://protonmail.com/support name ProtonMail vendor "Proton Technologies AG")

5 LIST "" "*"
0 `

**Comment by bengotow** [comment]
Hey folks! @Nelyah's original issue with `UIDVALIDITY` was fixed in 1.0.9 - I think this issue with ProtonMail is unrelated and should probably go in a separate thread. Offhand, it looks like Protonmail is telling Mailspring that there are no selectable folders in the account at all, which might confuse it into thinking auth failed or crash the client (will double check.) You might try adding at least one folder (like an INBOX) to your account and trying to auth again. (I'm not familiar with ProtonMail Bridge but not having any results from `LIST` seems really strange.)
